lib/streamlit/git_util.py

```diff
@@ -61,13 +61,46 @@ def is_valid(self) -> bool:
     def tracking_branch(self):
         if not self.is_valid():
             return None
+
+        if self.is_head_detached:
+            return None
+
         return self.repo.active_branch.tracking_branch()
 
+    @property
+    def untracked_files(self):
+        return self.repo.untracked_files
+
+    @property
+    def is_head_detached(self):
+        return self.repo.head.is_detached
```

Comment:
nit: should these have `if not self.is_valid():` checks like the other methods in this class?